### ISSUE TITLE
shapley kernel proof (nips supplement)

### ISSUE THREAD

**Issue by emilyjia** [shap/shap#1054]
Do you have more detail for the proof? Through some simulations I get that

$$X^TWX = \frac{M}{M-1}I + cJ$$ instead of $$X^TWX = \frac{1}{M-1}I + cJ$$

(The inverse in the paper is that of the correctly computed X^TWX.) Do you know if there is a not-bashy mathematical argument to see that X^TWX takes that form M/M-1 I + cJ?

**Comment by davidrosenberg** [comment]
There's a derivation of a more general, elegant result in a paper by Charnes et. al. in 1988, which defines generalized Shapley values. It basically gives a characterization of [generalized] Shapley values as the solution to an optimization problem -- approximating one set function by another simpler one. The notation is hard to follow (and/or there are a lot of typos/bugs?), but there's a very helpful blogpost [here](https://ypei.me/posts/2018-12-02-lime-shapley.html) that goes through the proof and is easier to follow.  Then the form of the Shapley kernel is a special case for the Shapley weight function.   I found this perspective pretty enlightening.

**Comment by Hideki105** [comment]
I cannot follow the discussion. Please tell me mathematical discussion in detail !

[Question](https://math.stackexchange.com/questions/4768867/shapley-kernel-proof?noredirect=1#comment10128945_4768867)

**Comment by Hideki105** [comment]
I think there is typographical error in the NIPS supplemental paper.

```math
X^TWX = \dfrac{1}{M-1}I+cJ \ \ (\mathrm{NIPS})
```

```math
X^TWX = \dfrac{M-1}{M}I+cJ \ \ (\mathrm{Update})
\tag{1}
```

If we suppose equation (1), using sherman-morrison formula, we can get  equation (2).
```math
(X^T WX)^{-1}=\dfrac{M}{M-1}I-\dfrac{\dfrac{M}{M-1}\dfrac{M}{M-1}cJ}{1+\dfrac{M}{M-1}cM}\rightarrow\dfrac{M}{M-1}I-\dfrac{1}{M-1}J=I+\dfrac{1}{M-1}(I-J) \ \ (c\rightarrow\infty)
\tag{2}
```
